duckduckgo/BrowserServicesKit
# Support JS pixels

Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/1199230911884351/1203393908384122/f
iOS PR:  
macOS PR: 
What kind of version bump will this require?: Major

**Optional**:

Tech Design URL:
CC:

**Description**:

This PR adds support for the new `sendJSPixel` Autofill event. This allows the JS layer to send pixels via the native app. These pixels do not include the platform name, so it's up to each platform to prepend that.



**Steps to test this PR**:

macOS:

1. Run the macOS PR
2. Add an Identity to the Autofill database
3. Visit a page with an address form, like the Amazon account page
4. Autofill your identity, and check that you see the `m.mac.autofill.identity` pixel fired

iOS:

We don't support identities on iOS, so to check this PR then test that the Autofill feature works with logins without firing a JS pixel. The only pixel fired from the JS side if for identities right now, so you should definitely not see this.



**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
